### PR TITLE
Fix formatting when the line before Field invocation contains whitespaces

### PR DESCRIPTION
**Pull request by gao-artur**
This PR fixes the second issue reported [here](https://github.com/graphql-dotnet/graphql-dotnet/pull/3720#issuecomment-1764981528)
image

The fix for `);` on a separate line is more complicated, so I'll leave it as is.

### CODE REVIEW

**Comment by gao-artur** [comment]
There is a nice VS extension to visualize trailing whitespaces in the code. It also allows to automatically remove them on document save. Unfortunately, it also removes whitespaces from the raw strings, so I had to disable the auto-remove feature to write this test.
https://marketplace.visualstudio.com/items?itemName=MadsKristensen.TrailingWhitespaceVisualizer

File: src/GraphQL.Analyzers.Tests/FieldBuilderAnalyzerTests.cs [hunk 1]
```diff
@@ -454,6 +454,55 @@ public MyGraphType()
...
+                        .Resolve(context => Resolve(
+                            "text"
+                        ));
+                }
+
+                public string Resolve(string s1) => s1;
+            }
+            """;
+
+        var expected = VerifyCS.Diagnostic(FieldBuilderAnalyzer.DoNotUseObsoleteFieldMethods).WithSpan(11, 9, 14, 15);
```
Comment (Shane32) [inline src/GraphQL.Analyzers.Tests/FieldBuilderAnalyzerTests.cs:502]:
```suggestion
        // ensure whitespace was not removed from input sample
        source.Replace("\r", "").ShouldContain("// next line contains whitespaces\n    ");

        var expected = VerifyCS.Diagnostic(FieldBuilderAnalyzer.DoNotUseObsoleteFieldMethods).WithSpan(11, 9, 14, 15);
```

**Comment by Shane32** [comment]
I looked back at my code, and although there isn't whitespace on the end of the previous line, there is a comment there, which is likely equivalent to whitespace:

image

In addition to the test you just added, perhaps we also add a similar test with a comment on the prior line instead of only whitespace.

File: src/GraphQL.Analyzers.Tests/FieldBuilderAnalyzerTests.cs (hunk 1 above)
Comment (Shane32) [reply src/GraphQL.Analyzers.Tests/FieldBuilderAnalyzerTests.cs:502]: I'm thinking that it is likely enough that some other formatting tool in the future may remove this whitespace, and the test should still pass.  So with a couple extra lines of code we can ensure the test is correct.

**Comment by Shane32** [comment]
> The fix for `);` on a separate line is more complicated, so I'll leave it as is.

:+1:

File: src/GraphQL.Analyzers.Tests/FieldBuilderAnalyzerTests.cs (hunk 1 above)
Comment (gao-artur) [reply src/GraphQL.Analyzers.Tests/FieldBuilderAnalyzerTests.cs:505]: Makes sense.

**Review by Shane32 — APPROVED** [review]

**Comment by Shane32** [comment]
Tested on my codebase; works great!
